specs/approved/power-control-intermediary.rst

```diff
@@ -0,0 +1,741 @@
+      the message body, or if the field has an invalid value.
+    * 403 Forbidden will be returned if the ``X-Auth-Token`` field in the
+      header is either absent or invalid for the resource being accessed.
+    * 404 Not Found will be returned if the identifier specified does not
+      correspond to a legitimate node UUID.
+    * 409 NodeLocked/ClientError is an error code specified in the Bare Metal
+      API call this request is proxied to. [7]_ The body of a 409 response will
+      be the same as that which was recieved from the Bare Metal API.
+    * 500 Internal Server Error will be returned if the internal request to the
+      Bare Metal service could not be fufilled.
+    * 503 NoFreeConductorWorkers is an error code specified in the Bare Metal
+      API call this request is proxied to. [7]_ The body of a 503 response will
+      be the same as that which was recieved from the Bare Metal API.
+
+  * Example Request::
+
+      X-Auth-Token: super-duper-secret-aaaaaaaaaaaa
+
+      {
+          "ResetType": "ForceOff"
+      }
+
+  +-----------+--------+----------------------------------------------+
+  | Name      | Type   | Description                                  |
+  +===========+========+==============================================+
+  | ResetType | string | The type of Reset action to take (see above) |
+  +-----------+--------+----------------------------------------------+
+
+.. [*] This is included for compatibility and should always be "OK", although
+       the Redfish schema allows for "Warning" and "Critical" as well.
+.. [*] This could be implemented, but it would come at the cost of running an
+       expensive service to expire active Sessions. This spec currently calls
+       for the max value (86400s) to be served to the user exclusively for the
+       sake of compatibility.
+.. [*] Five power states are possible for an Ironic node: "POWER_ON",
+       "POWER_OFF", "SOFT_POWER_OFF", "REBOOT", and "SOFT_REBOOT". Three of
+       these map cleanly onto defined Redfish power states (POWER_ON -> On,
+       POWER_OFF -> Off, SOFT_POWER_OFF -> Powering Off), but the Ironic
+       "REBOOT" power states do not. I see a few possible solutions here, one
+       being to map both "reboot" states onto "Powering On", the other being
+       to serve a custom schema that includes a "Rebooting" state. If the
+       latter is used, we could either have both Ironic "REBOOT" states map on
+       to this new Redfish state, or we could have one map onto the new state,
+       and the other map onto "Powering On".
+.. [*] The Redfish schema for ResetType also includes "Nmi" (diagnostic
+       interrupt) and "PushPowerButton" (simulates a physical power button
+       press event) but since these are not part of the Ironic spec, they are
+       made unavailable here.
+
+Client (CLI) impact
+-------------------
+None.
+
+"openstack baremetal" CLI
+~~~~~~~~~~~~~~~~~~~~~~~~~
+None.
```

Comment:
Maybe add a brief explanation as to why this Ironic REST API addition doesn't require a corresponding updated to the CLI (because users can just use the existing Ironic commands)? Might not be necessary.

Reply:
extra detail can't hurt, added!